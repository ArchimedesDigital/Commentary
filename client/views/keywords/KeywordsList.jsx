KeywordsList = React.createClass({

  // This mixin makes the getMeteorData method keyword
  mixins: [ReactMeteorData],

  propTypes: {
  },

  // Loads items from the keywords collection and puts them on this.data.keywords
  getMeteorData() {
    let query = {};

    return {
      keywords: Keywords.find(query, {sort: {title: 1}}).fetch()
    };
  },

  renderKeywords() {

    return this.data.keywords.map((keyword, i) => {
      return <KeywordTeaser
              key={i}
              keyword={keyword} />;

    });

  },

  render() {

     return (
       <div className="keywords-list">
         {this.renderKeywords()}

       </div>


      );
    }


});
