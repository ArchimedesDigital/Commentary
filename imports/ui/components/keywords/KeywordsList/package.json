{
  "name": "KeywordsList",
  "version": "0.0.1",
  "private": true,
  "main": "./KeywordsList.js"
}
