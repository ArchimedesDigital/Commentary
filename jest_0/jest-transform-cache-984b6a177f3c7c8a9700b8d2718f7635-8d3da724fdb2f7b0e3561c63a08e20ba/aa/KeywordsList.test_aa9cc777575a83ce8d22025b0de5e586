/* istanbul ignore next */'use strict';var /* istanbul ignore next */_react = require('react'); /* istanbul ignore next */var _react2 = _interopRequireDefault(_react);
var /* istanbul ignore next */_reactTestRenderer = require('react-test-renderer'); /* istanbul ignore next */var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);


var /* istanbul ignore next */_KeywordsList = require('./KeywordsList'); /* istanbul ignore next */var _KeywordsList2 = _interopRequireDefault(_KeywordsList);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

describe('KeywordsList', function () {
	it('renders correctly', function () {

		var tree = /* istanbul ignore next */_reactTestRenderer2.default.
		create(
		/* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_KeywordsList2.default, /* istanbul ignore next */{
			type: 'word' })).


		toJSON();
		expect(tree).toMatchSnapshot();
	});
}); // component: