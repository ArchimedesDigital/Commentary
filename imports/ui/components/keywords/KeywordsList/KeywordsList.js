import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Meteor } from 'meteor/meteor';

import { compose } from 'react-apollo';

// models
import Keywords from '/imports/models/keywords';

// graphql
import { keywordsQuery } from '/imports/graphql/methods/keywords';

// components
import KeywordTeaser from '/imports/ui/components/keywords/KeywordTeaser';

function getKeywordsByQuery(query, limit) {

	if (query.loading) {
		return [];
	}
	return query.keywords.slice(0, limit);
}

class KeywordsList extends Component {

	renderKeywords() {
		const { keywords } = this.state;

		if (!keywords) {
			return null;
		}

		return keywords.map((keyword, i) => (
			<KeywordTeaser
				key={i}
				keyword={keyword}
			/>
		));
	}
	constructor(props) {
		super(props);
		this.state = {};
	}
	componentWillReceiveProps(newProps) {
		const { type, limit } = newProps;
		const skip = 0;
		const tenantId = sessionStorage.getItem('tenantId');
		let _limit = 100;
		const query = {
			type: type,
			count: { $gte: 1}
		};
		if (limit) {
			_limit = limit;
		}
		newProps.keywordsQuery.refetch({
			tenantId: tenantId,
			queryParam: JSON.stringify(query)
		});
	
		let keywords = [];
		switch (type) {
		case 'word':
			keywords = getKeywordsByQuery(newProps.keywordsQuery, _limit);
			break;
		case 'idea':
			keywords = getKeywordsByQuery(newProps.keywordsQuery, _limit);
			break;
		default:
			break;
		}
		this.setState({
			keywords: keywords,
		});
	}

	render() {
		return (
			<div className="keywords-list">
				{this.renderKeywords()}
			</div>
		);
	}
}

KeywordsList.propTypes = {
	type: PropTypes.string.isRequired,
	limit: PropTypes.number,
	keywords: PropTypes.array,
};

export default compose(keywordsQuery)(KeywordsList);
