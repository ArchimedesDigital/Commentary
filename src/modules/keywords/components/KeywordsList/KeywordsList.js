import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { compose } from 'react-apollo';

// graphql
import { keywordsQuery } from '../../graphql/queries/keywords';

// components
import KeywordTeaser from '../KeywordsTeaser/KeywordTeaser';

function getKeywordsByQuery(query, limit) {

	if (query.loading) {
		return [];
	}
	return query.keywords.slice(0, limit);
}

class KeywordsList extends Component {
	constructor(props) {
		super(props);
		this.state = {};
		const tenantId = sessionStorage.getItem('tenantId');

		const query = {
			type: this.props.type,
			count: { $gte: 1}
		};
		this.props.keywordsQuery.refetch({
			tenantId: tenantId,
			queryParam: JSON.stringify(query)
		});
	}

	renderKeywords() {
		const { keywords } = this.state;

		if (!keywords) {
			return null;
		}

		return keywords.map((keyword, i) => (
			<KeywordTeaser
				key={i}
				keyword={keyword}
			/>
		));
	}

	componentWillReceiveProps(newProps) {
		const { type, limit } = newProps;
		let _limit = 100;

		let keywords = [];
		if (limit) {
			_limit = limit;
		}
		switch (type) {
		case 'word':
			keywords = getKeywordsByQuery(newProps.keywordsQuery, _limit);
			break;
		case 'idea':
			keywords = getKeywordsByQuery(newProps.keywordsQuery, _limit);
			break;
		default:
			break;
		}
		this.setState({
			keywords: keywords,
		});
	}

	render() {
		return (
			<div className="keywords-list">
				{this.renderKeywords()}
			</div>
		);
	}
}

KeywordsList.propTypes = {
	type: PropTypes.string.isRequired,
	limit: PropTypes.number,
	keywordsQuery: PropTypes.object,
};

export default compose(keywordsQuery)(KeywordsList);
