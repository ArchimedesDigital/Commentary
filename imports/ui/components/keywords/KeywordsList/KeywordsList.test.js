import React from 'react';
import renderer from 'react-test-renderer';

// component:
import KeywordsList from './KeywordsList';

describe('KeywordsList', () => {
	it('renders correctly', () => {

		const tree = renderer
			.create(
				<KeywordsList
					type="word"
				/>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
