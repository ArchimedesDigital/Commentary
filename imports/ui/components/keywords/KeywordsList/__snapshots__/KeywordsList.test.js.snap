// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeywordsList renders correctly 1`] = `
<div
  className="keywords-list"
/>
`;
